test/unit/algorithm_mode/test_callback.py

```diff
@@ -0,0 +1,161 @@
+import os
+import shutil
+import pathlib
+import tempfile
+import time
+import unittest
+from unittest.mock import patch
+from sagemaker_xgboost_container.algorithm_mode.callback import SaveCheckpoint
+
+
+class TestSaveCheckpoint(unittest.TestCase):
+
+    def setUp(self):
+
+        self.test_dir = tempfile.mkdtemp()
+
+    @patch("xgboost.core.CallbackEnv")
+    def test_not_master_node(self, env):
+
+        callback = SaveCheckpoint(checkpoint_dir=self.test_dir)
+
+        env.rank = 1
+        env.iteration = 10
+        callback(env)
+
+        self.assertEqual(len(os.listdir(self.test_dir)), 0)
+
+        callback.stop()
+
+    @patch("xgboost.core.CallbackEnv")
+    def test_SaveCheckpoint_single_iteration(self, env):
+
+        env.rank = 0
+        env.iteration = 42
+        env.end_iteration = 100
+
+        callback = SaveCheckpoint(checkpoint_dir=self.test_dir)
+        callback(env)
+
+        file_path = os.path.join(self.test_dir, "xgboost-checkpoint.000000000042")
+        self.assertTrue(os.path.isfile(file_path))
+        self.assertTrue(len(os.listdir(self.test_dir)), 1)
```

Comment:
I have some basic unit tests for marker files in `test_SaveCheckpoint_uploading`. Integration tests don't really check for marker files.

Reply:
On that note, it would be good to include the information you've provided in this PR as comments in the code so it's clear for another developer in the future what you've done.